### PR TITLE
wr-74: Create achievements table

### PR DESCRIPTION
**Pull request by AnastasiaR2**
1. Created migration to add the achievements table.
2. Prepopulated achievements table with common achievements (write the first article, write 5/10/15/25/50 articles, same with comments).
3. Created achievements repository\service\controller.

### CODE REVIEW

File: backend/src/packages/achievements/achievement.entity.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      name,
+      description,
+    });
+  }
+
+  public static initializeNew({
+    key,
+    name,
+    description,
+  }: Omit<AchievementEntityType, 'id'>): AchievementEntity {
```
Comment (NazarKovtun1) [inline backend/src/packages/achievements/achievement.entity.ts]: I offer move this omit to separate type

type AchievementEntityPayloadType = Omit<AchievementEntityType, 'id'>
```suggestion
  }: AchievementEntityPayloadType): AchievementEntity {
```

Comment (NazarKovtun1) [reply backend/src/packages/achievements/achievement.entity.ts]: Move this type before AchievmentEntity class but after imports

File: backend/package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
     "migrate:dev:make": "node --loader ts-paths-esm-loader ../node_modules/knex/bin/cli.js migrate:make -x ts",
     "migrate:dev:down": "node --loader ts-paths-esm-loader ../node_modules/knex/bin/cli.js migrate:down",
     "migrate:dev:rollback": "node --loader ts-paths-esm-loader ../node_modules/knex/bin/cli.js migrate:rollback --all",
+    "seed:dev": "node --loader ts-paths-esm-loader ../node_modules/knex/bin/cli.js seed:run",
```
Comment (andanf-e) [inline backend/package.json]: We don't need it
```suggestion

```

File: backend/src/db/seeds/achievements-seed.ts
Comment (andanf-e) [inline backend/src/db/seeds/achievements-seed.ts:1]: We don't need it

File: backend/src/libs/packages/database/database.package.ts [hunk 3]
```diff
@@ -43,6 +43,9 @@ class Database implements IDatabase {
         directory: 'src/db/migrations',
         tableName: DatabaseTableName.MIGRATIONS,
       },
+      seeds: {
+        directory: 'src/db/seeds',
+      },
```
Comment (andanf-e) [inline backend/src/libs/packages/database/database.package.ts]: Move data prepopulation to `migrations`
```suggestion

```

**Review by andanf-e — CHANGES_REQUESTED** [review]

**Review by andanf-e — DISMISSED** [review]

**Review by mhrechkin — DISMISSED** [review]

**Review by NazarKovtun1 — DISMISSED** [review]

File: shared/src/packages/achievements/libs/types/achievement-entity.type.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+type AchievementEntityType = {
```
Comment (andanf-e) [inline shared/src/packages/achievements/libs/types/achievement-entity.type.ts]: type naming should not contain Type word itself

**Review by andanf-e — CHANGES_REQUESTED** [review]

**Review by andanf-e — APPROVED** [review]

**Review by NazarKovtun1 — APPROVED** [review]

**Pull request merged**